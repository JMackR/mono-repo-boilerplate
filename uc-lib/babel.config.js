module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
