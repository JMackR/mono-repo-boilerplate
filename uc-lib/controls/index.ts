export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
