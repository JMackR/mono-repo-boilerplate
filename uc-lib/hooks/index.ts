export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
