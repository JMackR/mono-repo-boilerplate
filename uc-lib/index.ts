export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
