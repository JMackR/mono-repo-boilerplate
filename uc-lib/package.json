{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
