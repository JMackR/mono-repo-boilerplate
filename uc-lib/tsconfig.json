{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
