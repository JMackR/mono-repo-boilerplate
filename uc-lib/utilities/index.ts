export * from "./dimensions-utility"
export * from "./media-query"
