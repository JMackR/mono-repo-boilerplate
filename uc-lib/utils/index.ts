export * from "./iif"
export * from "./format-number"
