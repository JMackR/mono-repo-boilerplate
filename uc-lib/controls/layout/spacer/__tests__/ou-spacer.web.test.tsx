import React from "react"
import * as renderer from "react-test-renderer"
import { Spacer, Text } from "uc-lib"

describe("Spacer Snapshot Tests", () => {
  test("Spacer renders horizontal space correctly", () => {
    const tree = renderer.create(
      <div style={{ flexDirection: "row" }}>
        <Text>Example String 1</Text>
        <Spacer sizeStep={2} direction="row" />
        <Text>Example String 2</Text>
      </div>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("Spacer renders vertical space correctly", () => {
    const tree = renderer.create(
      <div style={{ flexDirection: "column" }}>
        <Text>Example String 1</Text>
        <Spacer sizeStep={2} direction="column" />
        <Text>Example String 2</Text>
      </div>,
    )
    expect(tree).toMatchSnapshot()
  })
})
