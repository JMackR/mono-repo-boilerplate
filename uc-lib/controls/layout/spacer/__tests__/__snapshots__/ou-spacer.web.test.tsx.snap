// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer Snapshot Tests Spacer renders horizontal space correctly 1`] = `
.text-textStyle {
  color: #121212;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.root {
  height: undefined;
  width: 8px;
  display: inline-block;
}

<div
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <div
    className="text-textStyle"
  >
    Example String 1
  </div>
  <div
    className="root"
  />
  <div
    className="text-textStyle"
  >
    Example String 2
  </div>
</div>
`;

exports[`Spacer Snapshot Tests Spacer renders vertical space correctly 1`] = `
.text-textStyle {
  color: #121212;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.root {
  height: 8px;
  width: undefined;
  display: block;
}

<div
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <div
    className="text-textStyle"
  >
    Example String 1
  </div>
  <div
    className="root"
  />
  <div
    className="text-textStyle"
  >
    Example String 2
  </div>
</div>
`;
